src/runtime/particle-spec.ts

```diff
@@ -367,23 +357,25 @@ export class ParticleSpec {
     return this.toString();
   }
 
-  private validateTrustClaims(claims: SerializedParticleTrustClaimSpec[]): Map<string, string> {
-    const results: Map<string, string> = new Map();
+  private validateTrustClaims(claims: ParticleTrustClaim[]): Map<string, ParticleTrustClaim> {
+    const results: Map<string, ParticleTrustClaim> = new Map();
     if (claims) {
       claims.forEach(claim => {
+        assert(!results.has(claim.handle), `Can't make multiple claims on the same output (${claim.handle}).`);
         assert(this.handleConnectionMap.has(claim.handle), `Can't make a claim on unknown handle ${claim.handle}.`);
         const handle = this.handleConnectionMap.get(claim.handle);
         assert(handle.isOutput, `Can't make a claim on handle ${claim.handle} (not an output handle).`);
-        results.set(claim.handle, claim.trustTag);
+        results.set(claim.handle, claim);
       });
     }
     return results;
   }
 
-  private validateTrustChecks(checks?: SerializedParticleTrustCheckSpec[]): Map<string, string[]> {
+  private validateTrustChecks(checks?: ParticleTrustCheck[]): Map<string, string[]> {
     const results: Map<string, string[]> = new Map();
     if (checks) {
       checks.forEach(check => {
+        assert(!results.has(check.handle), `Can't make multiple checks on the same input (${check.handle}).`);
```

Comment:
And again. You should really check all of the asserts in this file.

Reply:
Same here. There's a few others asserts in this file, but I'm not sure if they're supposed to be for internal errors or manifest errors... I've just left them alone.